pkg/riff/commands/doctor.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/projectriff/riff/pkg/cli"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DoctorOptions struct {
+}
+
+var RequiredNamespaces = []string{
+	"istio-system",
+	"knative-build",
+	"knative-serving",
+	"riff-system",
+}
+
+func (opts *DoctorOptions) Validate(ctx context.Context) *cli.FieldError {
+	return cli.EmptyFieldError
+}
+
+func (opts *DoctorOptions) Exec(ctx context.Context, c *cli.Config) error {
+	missingNamespaces, err := checkMissingNamespaces(c)
+
+	if err != nil {
+		c.Errorf(err.Error())
```

Comment:
No need to log an error and return it. Logging the error is handled by the caller. By logging and returning the error will be printed on the console twice.